build-tools/webpack/assets-writer-plugin.js

```diff
@@ -43,50 +43,41 @@ Object.assign( AssetsWriter.prototype, {
 				reasons: false,
 			} );
 
+			function fixupPath( f ) {
+				return path.join( stats.publicPath, f );
+			}
+
+			// Exclude hot update files (info.hotModuleReplacement) and source maps (info.development)
+			function isDevelopmentAsset( name ) {
+				const asset = stats.assets.find( ( a ) => a.name === name );
+				if ( ! asset ) {
+					return false;
+				}
+
+				return asset.info.hotModuleReplacement || asset.info.development;
+			}
```

Comment:
This fixes one of possible causes of the "`window.AppBoot` is not a function" error. In development mode, we were writing the `*.hot-update.*` assets to the `assets.json` file, and the dev server was probably serving them as one of the scripts in the initial HTML page. That's wrong, hot updates are supposed to be sent only to a live page without reload. Once you reload, you'll load the updated asset.

See how `html-webpack-plugin` does exactly the same filtering of assets that will be included on the HTML page: https://github.com/jantimon/html-webpack-plugin/blob/7299866767d7256c65e1d11daaed83426b6d74aa/index.js#L558-L564

Reply:
Nice find 👍